### PR TITLE
Update broken links in CONTRIBUTING.md

### LINKED ISSUE #3836: Link to the review process on CONTRIBUTING.md is broken
### Description

It seems like the link to [the review process we use for simple changes](https://docs.docker.com/opensource/workflow/make-a-contribution/) on this [doc](https://github.com/docker/cli/blob/master/CONTRIBUTING.md#quick-contribution-tips-and-guidelines) is broken.

### Reproduce

View the doc for quick contribution tips and guidelines and click the link to `the review process we use for simple changes`.

### Expected behavior

On clicking the link, the page on contribution guidelines should be opened.

### docker version

```bash
-
```

### docker info

```bash
-
```

### Additional Info

_No response_

### LINKED ISSUE #5416: Broken Links in Contributing.md
### Description

Below Links are broken in [CONTRIBUTING.md](https://github.com/docker/cli/blob/master/CONTRIBUTING.md)

- [the review process we use for simple changes](https://docs.docker.com/opensource/workflow/make-a-contribution/) under [Pull requests are always welcome](https://github.com/docker/cli/blob/master/CONTRIBUTING.md#pull-requests-are-always-welcome).
- [our style guide](https://docs.docker.com/opensource/doc-style) under [Conventions](https://github.com/docker/cli/blob/master/CONTRIBUTING.md#conventions).
- [building the documentation](https://docs.docker.com/opensource/project/test-and-docs/#build-and-test-the-documentation) under [Conventions](https://github.com/docker/cli/blob/master/CONTRIBUTING.md#conventions).

Also FYI [golint](https://github.com/golang/lint) (under [Coding Style](https://github.com/docker/cli/blob/master/CONTRIBUTING.md#coding-style)) is archived.

Apologize for raising it as a bug, could not find a issue template related to documentation related issues.

### Reproduce

Not Applicable

### Expected behavior

Not Applicable

### docker version

```bash
Not Applicable
```

### docker info

```bash
Not Applicable
```

### Additional Info

Not Applicable

### PR DESCRIPTION
**Pull request by DilepDev**
- fixes https://github.com/docker/cli/issues/3836

**- What I did**
In CONTRIBUTING.md updated "the review process we use for simple changes", "our style guide" and "building the documentation" with working links.

**- How I did it**
Used google and [docs.docker](https://docs.docker.com/) for information

**- How to verify it**
NA

**- Description for the changelog**

```markdown changelog
In CONTRIBUTING.md "the review process we use for simple changes", "our style guide" and "building the documentation" will point to the correct working links
```

**- A picture of a cute animal (not mandatory but encouraged)**

tiger

"closes #5416 "

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -66,7 +66,7 @@ anybody starts working on it.
 We are always thrilled to receive pull requests. We do our best to process them
 quickly. If your pull request is not accepted on the first try,
 don't get discouraged! Our contributor's guide explains [the review process we
-use for simple changes](https://docs.docker.com/opensource/workflow/make-a-contribution/).
+use for simple changes](https://github.com/docker/docker/blob/master/project/REVIEWING.md).
```
Comment (laurazard) [inline CONTRIBUTING.md:69]: For other reviewers, I think this is a version of what this page was supposed to point at – https://web.archive.org/web/20160305022033/https://docs.docker.com/opensource/workflow/make-a-contribution/

**Comment by Benehiko** [comment]
Similar to https://github.com/docker/cli/pull/5041

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -124,8 +124,8 @@ submitting a pull request.
 Update the documentation when creating or modifying features. Test your
 documentation changes for clarity, concision, and correctness, as well as a
 clean documentation build. See our contributors guide for [our style
-guide](https://docs.docker.com/opensource/doc-style) and instructions on [building
-the documentation](https://docs.docker.com/opensource/project/test-and-docs/#build-and-test-the-documentation).
+guide](https://docs.docker.com/contribute/style/grammar/) and instructions on [building
```
Comment (laurazard) [inline CONTRIBUTING.md:127]: Same here: https://web.archive.org/web/20160305023756/https://docs.docker.com/opensource/doc-style/

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -124,8 +124,8 @@ submitting a pull request.
...
+the documentation](https://docs.docker.com/contribute/).
```
Comment (laurazard) [inline CONTRIBUTING.md:128]: And here: https://web.archive.org/web/20151209160310/http://docs.docker.com/opensource/project/test-and-docs/

File: CONTRIBUTING.md (hunk 1 above)
Comment (laurazard) [reply CONTRIBUTING.md:69]: (I wonder if we still have something like these. These docs were good, and the replacements – although seemingly the closest we currently have – don't cover most of what they used to cover)

Comment (thaJeztah) [reply CONTRIBUTING.md:69]: I think most of what was left in the docs repo was moved back to https://github.com/moby/moby/tree/master/docs/contributing

Comment (laurazard) [reply CONTRIBUTING.md:69]: Yeah but did you see the link I posted? We don't really have a comparable doc right now – we have technical docs on how to debug daemon, setup environment, etc. but this had 
Screenshot 2024-09-09 at 12 18 20

**Comment by DilepDev** [comment]
Thank you @laurazard, @thaJeztah for the inputs, what would you suggest next step to be, is it to create the missing documentation in `docker/cli` referencing [moby/docs](https://github.com/moby/moby/tree/master/docs/contributing) and links from the waybackmachine ?

**Comment by laurazard** [comment]
@DilepDev This PR is still an improvement overall, so I definitely think we should still merge it, and we can think about moving/maybe bringing back some of those old docs into moby/moby or here later.

**Review by laurazard — APPROVED** [review]

**Review by thaJeztah — APPROVED** [review]
LGTM

**Pull request merged**